.github/workflows/fetch-bank-holidays.yml

```diff
@@ -0,0 +1,50 @@
+name: Fetch bank holidays JSON from the gov.uk API
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  call_api:
+    name: Call API
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.call_api.outputs.CHANGES }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Call API
+        id: call_api
+        run: |
+          curl -o bank-holidays.json https://www.gov.uk/bank-holidays.json
+          if diff ./server/data/bankHolidays/bank-holidays.json bank-holidays.json; then
+            echo "Files are the same"
+          else
+            echo "Files are different"
+            mv bank-holidays.json ./server/data/bankHolidays/bank-holidays.json
+            echo "CHANGES=true" >> $GITHUB_OUTPUT
+          fi
```

Comment:
I think you can get away with not doing this actually - if you look at the [Generate Types](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/main/.github/workflows/generate-types.yml) action, and the [runs](https://github.com/ministryofjustice/hmpps-approved-premises-ui/actions/workflows/generate-types.yml), it runs every time it gets triggered, but only opens a PR if there are any changes

Reply:
that would make things a lot simpler. 
I read something in the docs that suggested Actions will create an empty PR but I think the `peter-evans/create-pull-request@v5` must be a bit smarter